packages/dd-trace/src/profiling/profiler.js

```diff
@@ -20,6 +19,11 @@ class Profiler extends EventEmitter {
     this._config = undefined
     this._timer = undefined
     this._lastStart = undefined
+
+    this._isServerless = !!process.env.AWS_LAMBDA_FUNCTION_NAME
+    this._profiledIntervals = 0
```

Comment:
Is this needed? It seems to just grab profiles faster until the flush interval is reached, then reset, and then collect faster again, but the lower flush interval would already achieve this.

A few questions:

- Would that negatively impact the application? I remember stopping and starting the profiler being the most expensive part of the profiler. Maybe @Qard can clarify if that's no longer the case.
- If we're able to do this efficiently for serverless, why can't we just always do this instead to simplify the code?
- A separate code path would be clearer. It took me a good 10 minutes to figure out what this actually does.